File: DS/Array/CPP/A5.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int countOccurance(vector<string> list,string str) {
    int count = 0;
    for(int i=0;i<list.size();i++) {
        if(list.at(i)==str) {
            count++;
        }
    }
    return count;
}

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */ 
    int N;
    cin>>N;
    vector<string> list;
    vector<string> query;
    for(int i=0;i<N;i++) {
        string temp;
        cin>>temp;
        list.push_back(temp);
        
    }
    int Q;
    cin>>Q;
    int Total[Q];
    for(int i=0;i<Q;i++) {
        string temp;
        cin>>temp;
        query.push_back(temp);
        
    }
    for(int i =0;i<Q;i++) {
        Total[i] = countOccurance(list,query.at(i));
        cout<<Total[i]<<endl;
    }

    return 0;
}

File: DS/Array/JS/A5.js
function countOccurance(list,str) {
    var count = 0;
    for(var i=0;i<list.length;i++) {
        if(list[i]==str) {
            count++;
        }
    }
    return count;
}

function processData(input) {
    //Enter your code here
    var line = input.split('\n');
    var N = parseInt(line[0])
    var Q = parseInt(line[N+1]);
    var list = [];
    var query = [];
    for(var i=1;i<=N;i++) {
        list.push(line[i]);
    }
    for(var i=N+2;i<line.length;i++) {
        query.push(line[i]);
    }
    for(var i =0;i<Q;i++) {
       console.log(countOccurance(list,query[i]));
    }
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
